core/src/main/java/org/apache/iceberg/util/PropertyUtil.java

```diff
@@ -70,4 +72,16 @@ public static String propertyAsString(Map<String, String> properties,
     }
     return defaultValue;
   }
+
+  public static Map<String, String> propertiesWithPrefix(Map<String, String> properties,
+                                                         String prefix) {
```

Comment:
[question] since this is a public should we check `prefix != null` as replace requires the target CharSequence to be NonNull ?

Reply:
I agree with this suggestion.